features/transient.feature

```diff
@@ -191,20 +220,74 @@ Feature: Manage WordPress transient cache
       Warning: Transient with key "foo2" is not set.
       """
 
+    When I run `wp transient get foo3 --network`
+    Then STDOUT should be:
+      """
+      bar3
+      """
+
+    When I run `wp transient get foo4 --network`
+    Then STDOUT should be:
+      """
+      bar4
+      """
+
+    When I run `wp --url=example.com/foo transient get foo5 --network`
+    Then STDOUT should be:
+      """
+      bar5
+      """
+
+    When I run `wp --url=example.com/foo transient get foo6 --network`
+    Then STDOUT should be:
+      """
+      bar6
+      """
+
+    When I run `wp transient delete --all --network`
+    Then STDOUT should be:
+      """
+      Success: 4 transients deleted from the database.
+      """
+
     When I try `wp transient get foo3 --network`
     Then STDERR should be:
       """
       Warning: Transient with key "foo3" is not set.
       """
 
-    When I run `wp transient set foo bar 60`
-    Then STDOUT should be:
+    When I try `wp transient get foo4 --network`
+    Then STDERR should be:
       """
-      Success: Transient added.
+      Warning: Transient with key "foo4" is not set.
+      """
+
+    When I try `wp --url=example.com/foo transient get foo5 --network`
+    Then STDERR should be:
+      """
+      Warning: Transient with key "foo5" is not set.
       """
 
+    When I try `wp --url=example.com/foo transient get foo6 --network`
+    Then STDERR should be:
+      """
+      Warning: Transient with key "foo6" is not set.
+      """
+
+  Scenario: Deleting expired transients on multisite
+    Given a WP multisite install
+    And I run `wp site create --slug=foo`
+
+    When I run `wp transient set foo bar 60`
+    And I run `wp transient set foo2 bar2 60`
+    And I run `wp transient set foo3 bar3 60 --network`
+    And I run `wp transient set foo4 bar4 60 --network`
+    And I run `wp --url=example.com/foo transient set foo5 bar5 60 --network`
+    And I run `wp --url=example.com/foo transient set foo6 bar6 60 --network`
     # Change timeout to be in the past.
-    When I run `wp option update _transient_timeout_foo 1321009871`
+    And I run `wp option update _transient_timeout_foo 1321009871`
+    And I run `wp option update _site_transient_timeout_foo3 1321009871`
+    And I run `wp --url=example.com/foo option update _site_transient_timeout_foo5 1321009871`
```

Comment:
@schlessera Since this is a multisite the transient is stored in the site meta table. `wp site option` should do it.

Reply:
Okay, that leaves me now with this:
```
01. $ wp transient get foo3 --network

    Warning: Transient with key "foo3" is not set.
    cwd: /tmp/wp-cli-test-run-transient.feature.119-5bf665ee4f6613.35024913/
    run time: 0.43096494674683
    exit status: 0
    In step `When I run `wp transient get foo3 --network`'.     # vendor/wp-cli/wp-cli-tests/features/steps/when.php:29
    From scenario `Deleting expired transients on single site'. # features/transient.feature:119
    Of feature `Manage WordPress transient cache'.              # features/transient.feature

02. $ wp transient get foo3 --network

    Warning: Transient with key "foo3" is not set.
    cwd: /tmp/wp-cli-test-run-transient.feature.277-5bf665f4cceb22.27477243/
    run time: 0.45121192932129
    exit status: 0
    In step `When I run `wp transient get foo3 --network`'.     # vendor/wp-cli/wp-cli-tests/features/steps/when.php:29
    From scenario `Deleting expired transients on multisite'.   # features/transient.feature:277
    Of feature `Manage WordPress transient cache'.              # features/transient.feature
```

Reply:
At least the failure is consistent across single/multisite now! ;)

Reply:
That's expected because WP deletes transients if they are expired: https://github.com/WordPress/wordpress-develop/blob/a75d153eeeac9154c2576697adf721027e1d073e/src/wp-includes/option.php#L702-L706

Reply:
Would you say it makes sense to set `WP_INSTALLING` in this case to be able to test the proper separation?

Reply:
Or use `get_option()` for this check instead, maybe?

Reply:
Yep, that's what I did in https://github.com/wp-cli/cache-command/pull/42/commits/814750baa61b581874b17f1031b7778b216149a1.

Reply:
As for the rest, do you think the tests make sense as they are laid out now?

Reply:
Yep, they are looking good to me, nice work!